src/Common/Sitemap/Sitemap.php

```diff
@@ -0,0 +1,223 @@
+     * Create the sitemap.
+     *
+     * @param (siring) $mode       Valid mode (for only access inside class)
+     * @param (string) $url        Valid url.
+     * @param (string) $lastMod    Last modify.
+     * @param (float)  $priority   Priority.
+     * @param (string) $changeFreq changeFreq.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+    */
+    private function create($mode, $url, $lastMod = null, $priority = 0.5, $changeFreq = 'weekly')
+    {
+        [$lastMod, $priority, $changeFreq] = [$lastMod ?: $this->lastMod, $priority ?: $this->priority, $changeFreq ?: $this->changeFreq];
+        if(!in_array($changeFreq, $this->validFrequencies, true)) {
+            throw new \InvalidArgumentException("The value of changeFreq is not valid", 500);
+        }       
+        $raw = str_replace([':url', ':lastmod', ':changefreq',':priority'], [$url, $lastMod, $changeFreq, (float) $priority], $this->raw);
+        if ($mode === 'create') {
+            $fileH = new SitemapWriter($this->file, 'writeOnly');
+            $fileH->write(Sitemap::START.PHP_EOL);
+            $fileH->write($raw);
+            $fileH->write(PHP_EOL.Sitemap::END);
+        } elseif ($mode === 'append') {
+            $fileH = new SitemapWriter($this->file, 'readOnly');
+            $sitemapData = $fileH->read();
+            $fileH = new SitemapWriter($this->file, 'writeOverride');
+            $sitemapData = str_replace('</urlset>', '', $sitemapData);
+            $sitemapData = $sitemapData . $raw;
+            $fileH->write($sitemapData);
+            $fileH->write(PHP_EOL.Sitemap::END);
+        }
+        $fileH->close();
+
+        return true;
+    }
+
+    /**
+     * Add item to sitemap.
+     *
+     * @param (string) $url        Valid url.
+     * @param (string) $lastMod    Last modify.
+     * @param (float)  $priority   Priority.
+     * @param (string) $changeFreq changeFreq.
+     *
+     * @since 3.0.0
+     *
+     * @return void
+    */
+    public function addItem($url, $lastMod = null, $priority = 0.5, $changeFreq = 'weekly')
+    {
+        if ($this->has($this->file)) {
+            $this->appendItem($url, $lastMod, $priority, $changeFreq);
+        } else {
+            $this->create('create', $url, $lastMod, $priority, $changeFreq);
+        }
+    }
+
+    /**
+     * Append item to sitemap.
+     *
+     * @param (string) $url        Valid url.
+     * @param (string) $lastMod    Last modify.
+     * @param (float)  $priority   Priority.
+     * @param (string) $changeFreq changeFreq.
+     *
+     * @since 3.0.0
+     *
+     * @return void
+    */
+    private function appendItem($url, $lastMod, $priority, $changeFreq)
+    {
+        $this->create('append', $url, $lastMod, $priority, $changeFreq);
+    }
+
+    /**
+     * Determine whether the sitemap exists.
+     **
+     * @since 3.0.0
+     *
+     * @return bool
+    */
+    public function has()
+    {
+        return file_exists($this->file) ? true : false;
```

Comment:
This condition is not good because the `file_exists` should return bool value.

Please consider following code:

```php
......
return file_exists($this->file);
......
```

Reply:
Yes its return true so how this logics is wrong

i just test this 
```php
var_dump(file_exists(route()->public.'sitemap_index.xml') ? true : false);
```

Result if file exists => `bool(true)`
Result if file not exists => 'bool(false)`

Reply:
That's right and the `?:` syntax is useless.

Reply:
so you means i can just return 

```php

return file_exists($this->file);
```
Okay thanks

Reply:
https://github.com/zestframework/Zest_Framework/commit/c5f37051363570aeb98f255f6a4612dc03535602

Basically this pull was old and might be you saw code inconsity